### ISSUE TITLE
Remove internal `v1` module

### ISSUE THREAD

**Issue by flaviuvadan** [argoproj-labs/hera-workflows#41]
It is not necessary to version the internal modules of Hera for a few reasons:
- it makes semantic versioning more complicated than it needs to be (should Hera release a major version update if the underlying SDK changes? what if a single object has to change, such as the `Config` to modify the domain and `verify_ssl` flag?). Major package updates will reflect breaking changes, and the CHANGELOG will showcase what's available in a specific release that can be installed via PyPi;
- the introduction of `v2` would mean maintaining two internal version, or more in the future, and a public 3.x version potentially, which can be very confusing for users of all levels of experience. Hera needs to keep in mind that the purpose of it is to make it easy to adopt Argo Workflows. Understanding what a version of Hera means is critical for adoption and clarity/simplicity of installation
- reduce maintenance burden - when "v1" is conceptually eradicated, Hera can slowly move towards v2, rather than introduce a partially developed v2, which would not be a great experience for users (simplicity is key, so Hera should keep it simple)

**Comment by flaviuvadan** [comment]
@dixon1e and @bartcode - I made this decision in order to facilitate making the changes you requested. In its current state I realized that Hera would _force_ us to release a V2 in order to introduce the interface changes that your uses cases require. Removing v1 (and the concept of internal library versioning) introduces a breaking change in which we can safely deploy the necessary SSL verification changes you requested. I am opening a PR soon to remove `v1`. Then I will open a PR to adjust the `Configuration` object in a branch off of the v1 removal branch and tag you in it.
